packages/frontend/src/features/sqlRunner/components/visualizations/PieChart.tsx

```diff
@@ -0,0 +1,47 @@
+import { type PieChartSQLConfig } from '@lightdash/common';
+import { Center, Skeleton } from '@mantine/core';
+import EChartsReact, { type EChartsReactProps } from 'echarts-for-react';
+import { memo, type FC } from 'react';
+import { type ResultsAndColumns } from '../../hooks/useSqlQueryRun';
+import { usePieChart } from '../../transformers/usePieChart';
+
+type PieChartProps = {
+    data: ResultsAndColumns;
+    config: PieChartSQLConfig;
+    isLoading: boolean;
+} & Partial<Pick<EChartsReactProps, 'style'>>;
+
+const PieChart: FC<PieChartProps> = memo(
+    ({ data, config, style, isLoading: isLoadingProp }) => {
+        const {
+            loading: transformLoading,
+            error,
+            value: spec,
+        } = usePieChart(data.results, data.columns, config);
+        const loading = isLoadingProp || transformLoading;
+
+        if (error) {
+            return <Center>{error.message}</Center>;
+        }
+
+        return (
+            <>
+                {!spec && <Skeleton h="100%" />}
+                {spec && (
+                    <EChartsReact
+                        option={spec}
+                        showLoading={loading}
+                        opts={{
+                            renderer: 'svg',
+                            width: 'auto',
+                            height: 'auto',
+                        }}
+                        style={style}
+                    />
+                )}
+            </>
+        );
+    },
+);
```

Comment:
I think it's evident that we could have a shared `EchartsReact` component that uses one unified hook that takes a `ChartKind`

Do you think we can have a `useCartesianChart` that calls the content of `useBarChart` or `usePieChart` depending on the type?
For example

```
<EchartsVisualization data config style type/>
```

then inside this component we have everything as is (similar to this PieChart), but we call a hook, like: `useEchartsVisualization` that calls different transformers (or the same transformer that creates a different spec depending on the type)

Reply:
Yea, I think it makes sense. Let me see if I hit any snags making this more like that.